File: Address.cpp
//
// Created by Bryan Lim on 25/9/24.
//

#include "Address.h"
#include "Block.h"

// Constructor for Address
Address::Address(Block* block, int index) : block(block), index(index) {}

// Getter for the Block object
Block* Address::getBlock() const {
    return block;
}

// Getter for the index within the Block
int Address::getIndex() const {
    return index;
}

File: Address.java
// class to represent which block a record is in and where in the block it is in

import java.io.Serializable;

public class Address implements Serializable {
    private Block block;
    private int block_index;
    private int index;

    public Address(Block block, int block_index, int index) {
        this.block = block;
        this.block_index = block_index;
        this.index = index;
    }

    public Block getBlock() {
        return block;
    }

    public int getBlockIndex() {
        return block_index;
    }

    public int getIndex() {
        return index;
    }
}
